src/psyclone/tests/psyir/nodes/omp_directives_test.py

```diff
@@ -848,6 +850,65 @@ def test_directive_infer_sharing_attributes(fortran_reader):
     assert list(sync)[0].name == 'k'
 
 
+def test_directive_infer_sharing_attributes_with_structures(fortran_reader):
+    ''' Tests for the _infer_sharing_attributes() method of OpenMP directives
+    with code that contains structure accesses.
+    '''
+    psyir = fortran_reader.psyir_from_source('''
+        subroutine my_subroutine()
+            use my_mod
+            integer :: i, scalar1
+            type(my_type) :: mt1, mt2
+            real, dimension(10) :: array
+            mt1%scalar1 = 3
+            do i = 1, 10
+               if (i .eq. 4) then
+                  mt2%field1%scalar1 = array(i)
+               endif
+               scalar1 = mt2%field1%scalar1 + mt1%scalar1
+               array(i) = scalar1
+            enddo
+        end subroutine''')
+    omptrans = OMPParallelTrans()
+    routine = psyir.walk(Routine)[0]
+    omptrans.apply(routine.children)
+    directive = psyir.walk(OMPParallelDirective)[0]
+    pvars, fpvars, sync = directive._infer_sharing_attributes()
+    assert len(pvars) == 2
+    assert sorted(pvars, key=lambda x: x.name)[0].name == 'i'
+    assert sorted(pvars, key=lambda x: x.name)[1].name == 'scalar1'
```

Comment:
Should this be an `xfail` too as `scalar1` is actually a member of `mt1` so doesn't exist in isolation?

Reply:
No, scalar1 is also an independent scalar variable.

`mt1` (which has the scalar1 component) is shared because it is just written outside the loop (hence once)

Reply:
Oh yes, sorry, I completely missed that.